SharpPcap/LibPcap/PcapDevice.cs

```diff
@@ -586,6 +586,38 @@ public virtual void SendPacket(byte[] p, int size)
         /// <param name="size">The number of bytes to send</param>
         public abstract void SendPacket(ReadOnlySpan<byte> p);
 
+        #region Timestamp
+        /// <summary>
+        /// To set a device's timestamp resolution pass the desired setting in when opening the device
+        /// </summary>
+        /// <remarks>
+        /// The name mismatch between resolution and precision is intentional. To the end user
+        /// we use the more correct term of 'resolution' but we have to match the libpcap term of 'precision'
+        /// used in functions for proper pinvoke mapping.
+        /// </remarks>
+        public virtual TimestampResolution TimestampResolution
+        {
+            get
+            {
+                ThrowIfNotOpen("device is not open");
+                return (TimestampResolution)LibPcapSafeNativeMethods.pcap_get_tstamp_precision(PcapHandle);
+            }
+        }
+
+        public static string TimestampName(TimestampType number)
+        {
+            var ptr =  LibPcapSafeNativeMethods.pcap_tstamp_type_val_to_name((int)number);
+            return Marshal.PtrToStringAnsi(ptr);
+
+        }
+
+        public static string TimestampDescription(TimestampType number)
+        {
+            var ptr = LibPcapSafeNativeMethods.pcap_tstamp_type_val_to_description((int)number);
+            return Marshal.PtrToStringAnsi(ptr);
+        }
```

Comment:
I think it would be better to put `TimestampName(TimestampType number)` and `TimestampDescription(TimestampType number)` in `PcapClock` class instead (not public). thoughts?

Reply:
@kayoub5 in that case we'd have LibPcapLiveDevice (or PcapInterface) calling static methods in PcapClock to get values that we'd then pass back to PcapClock? That feels a bit circular no?

Reply:
My idea was that PcapClock will only have one argument in the constructor, the TimestampType, and the logic for getting the value of all other properties would be encapsulated in PcapClock